### ISSUE TITLE
Add an invitiation workflow

### ISSUE THREAD

**Issue by sissbruecker** [sissbruecker/linkding#42]
Currently there is no UI workflow to add new users to the system. The only options are creating new super-users using the Django script or enabling registration for a limited time. For self-hosted scenarios, it would make sense if the application would support invites. That would make it simple to add specific users (such as family members, friends, colleagues) without setting a password for them.

The workflow would look something like this:
- Application provides a page + form to invite a new user per mail
- User clicks on link in mail and lands on page / form where they can complete their registration details and set their password
- Completing the form enables the account and the user is now able to login

Would need to consider if this needs to be behind a config flag or if the invite feature can be active all the time (for super users).

**Comment by sissbruecker** [comment]
Rather than sending a mail automatically, it would be a lot easier to just provide an invite link that can be shared however the user likes. If you just want to invite like 3 people, sharing the 3 links is likely faster than having to figure out how to configure an SMTP service.

**Comment by Lanhild** [comment]
Agreed, as an admin, having the ability to create an invitation link that can expire in `x` hours, days... or after a specific number of uses. Very useful in large-scale use cases
